weave-js/src/components/PagePanelComponents/Home/Browse3/datasets/jsonUtils.ts

```diff
@@ -16,45 +19,53 @@ const unnestObject = (obj: any, prefix = ''): Record<string, any> => {
   return result;
 };
 
-export const parseJSON = async (file: File): Promise<ParseResult> => {
+export const parseJSON = async (file: File): Promise<JSONParseResult> => {
   return new Promise((resolve, reject) => {
     const reader = new FileReader();
     reader.onload = e => {
       try {
         const text = e.target?.result as string;
         const data = JSON.parse(text);
 
-        // Handle both array of objects and single object
-        const rows = Array.isArray(data) ? data : [data];
+        if (!Array.isArray(data)) {
+          if (typeof data === 'object' && data !== null) {
+            // Single object case - wrap in array
+            const rows = [data];
+            resolve({
+              data: rows.map(row => unnestObject(row)),
+            });
+          } else {
+            reject(
+              new Error(
```

Comment:
when can this happen?

Reply:
If the json is not an array or an object, ie the whole file is one scalar value, like `1` or `"hello"`